interfaces/apparmor/template.go

```diff
@@ -314,6 +314,9 @@ var templateCommon = `
   # Read-only of this snap
   /var/lib/snapd/snaps/@{SNAP_NAME}_*.snap r,
 
+  # Read-only of snapd restart state for snapctl specifically
+  /var/lib/snapd/maintenance.json r,
```

Comment:
Should `snap run` also read that file in the case it wants to wake up snapd to ensure system-key is set? Note that this does not require any additional permissions as `snap run` is unconfined.

Reply:
hmm I need to think about the implications of `snap run` waking up snapd to ensure system-key is set, let me read about how that works, but unless this change introduces a regression there, I would leave further improvements to followup PR's, as this PR is blocking uc20 1.0 now so less is more now :-)

Reply:
so I did eventually look into this and I think we are "okay" without any more changes here, but we could do things a bit better.

Mainly that with `snap run`, if a maintenance.json file exists, when `snap run` tries to ping snapd via the `system-info` endpoint, it will just eventually either timeout for a full system reboot, or it will get a new snapd around which can respond. The thing we could do better is we could issue a message saying that the user should wait for a reboot in that case if there is a maintenance.json but I don't think that's strictly necessary as it's probably unexpected that you are trying to run a snap app and it responds saying "system is rebooting", but maybe there is a good message we could do there. I will just add a TODO to that code to think about what that should look like